microservices/notification-service/src/notification/notification.service.ts

```diff
@@ -0,0 +1,48 @@
+import {Injectable} from "@nestjs/common";
+import type {
+    NotificationEndpoint, NotificationInput, NotificationOutput,
+} from "@sprocketbot/common";
+import {
+    BotEndpoint,
+    BotService,
+    config,
+    NotificationMessageType,
+    RedisService,
+} from "@sprocketbot/common";
+import {randomUUID} from "crypto";
+
+@Injectable()
+export class NotificationService {
+    private readonly prefix = `${config.redis.prefix}:notification:`;
+
+    constructor(
+        private readonly redisService: RedisService,
+        private readonly botService: BotService,
+    ) {}
+
+    async sendNotification(data: NotificationInput<NotificationEndpoint.SendNotification>): Promise<NotificationOutput<NotificationEndpoint.SendNotification>> {
+        if (data.payload) {
+            const notificationPayload = {
+                id: randomUUID(),
+                type: data.type,
+                expiration: data.expiration,
+                ...data.payload,
```

Comment:
```suggestion
                ...data.payload,
```

Reply:
`data.payload` doesn't contain type and expiration

Reply:
Am brick